dalite_xblock/passport_utils.py

```diff
@@ -0,0 +1,84 @@
+"""
+Module that handles passport parsing and generation.
+
+Separate module so I know no XBlock related imports are used from here, and
+it can be launched locally on vanilla python, just to generate passports.
+"""
+import base64
+from collections import namedtuple
+import logging
+
+logger = logging.getLogger(__name__)
+
+DaliteLtiPassport = namedtuple("DaliteLtiPassport", ["lti_id", "dalite_root_url", "lti_key", "lti_secret"])
+
+DALITE_PASSPORT_MARKER = "dalite-xblock"
+
+MALFORMED_LTI_PASSPORT_MESSAGE = u"Malformed Dalite-XBlock LTI Passport: %s - skipping"
+
+
+def prepare_passport(passport_data):
+    """
+    Create passport in a dalite-xblock format.
+
+    :param  DalitePassportData passport_data:
+    :returns: Dalite passport that can be pasted into studio
+    :rtype: str
+    """
+    decoded_data = ";".join((passport_data.dalite_root_url, passport_data.lti_key, passport_data.lti_secret))
+    encoded_part = base64.b64encode(decoded_data)
+    return ":".join((passport_data.lti_id, DALITE_PASSPORT_MARKER, encoded_part))
+
+
+def is_passport_for_dalite_xblock(passport_str):
+    """
+    Check if passport is applicable for dalite xblock.
+
+    :param str passport_str: A passport string.
+    :return: bool
+    """
+    passport_parts = passport_str.split(":")
+    if len(passport_parts) != 3 or passport_parts[1] != DALITE_PASSPORT_MARKER:
+        return False
+    encoded_passport = passport_parts[2]
+    try:
+        decoded_passport = base64.b64decode(encoded_passport)
+    except TypeError:
+        logger.warn(MALFORMED_LTI_PASSPORT_MESSAGE, passport_str)
+        return False
+    encoded_part_parts = decoded_passport.split(";")
+    if len(encoded_part_parts) != 3:
+        logger.warn(MALFORMED_LTI_PASSPORT_MESSAGE, passport_str)
+        return False
+    return True
+
+
+def parse_passport(passport_str):
+    """
+    Parse passport.
+
+    :param str passport_str: A passport string.
+    :rtype: DaliteLtiPassport
+    """
+    passport_parts = passport_str.split(":")
+    lti_id = passport_parts[0]
+    encoded_passport = passport_parts[2]
+    decoded_passport = base64.b64decode(encoded_passport)
+    dalite_root_url, lti_key, lti_secret = decoded_passport.split(";")
+    return DaliteLtiPassport(
+        lti_id=lti_id, lti_key=lti_key, lti_secret=lti_secret, dalite_root_url=dalite_root_url
+    )
+
```

Comment:
I was considering similar solution, but in the end I decided that having `parse_dalite_passport` always return a proper `DaliteLtiPassport` is worth additional hassle. However in this case I don't believe either option is significantly better than the other, it will remove some code. So I implemented it in 

Note that pylint doesn't like filter and map, so I just used list comprehensions.